controllers/mtls_reconciler.go

```diff
@@ -0,0 +1,307 @@
+
+func (r *MTLSReconciler) Subscription() *controller.Subscription {
+	return &controller.Subscription{ReconcileFunc: r.Run, Events: []controller.ResourceEventMatcher{
+		{Kind: ptr.To(kuadrantv1beta1.KuadrantGroupKind)},
+		{Kind: ptr.To(kuadrantv1.RateLimitPolicyGroupKind)},
+		{Kind: ptr.To(kuadrantv1.AuthPolicyGroupKind)},
+		{Kind: ptr.To(machinery.HTTPRouteGroupKind)},
+		{Kind: ptr.To(machinery.GatewayGroupKind)},
+		{Kind: ptr.To(istio.PeerAuthenticationGroupKind)},
+	},
+	}
+}
+
+//+kubebuilder:rbac:groups=security.istio.io,resources=peerauthenticators,verbs=get;list;watch;create;update;patch;delete
+
+func (r *MTLSReconciler) Run(eventCtx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, _ *sync.Map) error {
+
+	logger := controller.LoggerFromContext(eventCtx).WithName("MTLSReconciler")
+	logger.V(1).Info("reconciling mtls", "status", "started")
+	defer logger.V(1).Info("reconciling mtls", "status", "completed")
+
+	targetables := topology.Targetables()
+	// only gateway associated with gatewayclass of type istio
+	// path may be able to take list of gateways and list rules
+
+	// the path is from a specific gateway to HttpRouteRule
+	gateways := targetables.Items(func(o machinery.Object) bool {
+		gateway, ok := o.(*machinery.Gateway)
+		return ok && gateway.Spec.GatewayClassName == "istio"
+	})
+
+	// default to removal for envoyfilters
+	// if adding the configuration these will be updated later.
+	valueMap := map[string]interface{}{}
+	operation := v1alpha32.EnvoyFilter_Patch_REMOVE
+
+	httpRouteRules := targetables.Items(func(o machinery.Object) bool {
+		_, ok := o.(*machinery.HTTPRouteRule)
+		return ok
+	})
+	anyAttachedAuthorRateLimitPolicy := false
+	policies := make([]machinery.Policy, 0)
+outer:
+	for _, gateway := range gateways {
+		for _, httpRouteRule := range httpRouteRules {
+			paths := targetables.Paths(gateway, httpRouteRule)
+			for _, path := range paths {
+				policies = kuadrantv1.PoliciesInPath(path, func(policy machinery.Policy) bool {
+					if _, ok := policy.(*kuadrantv1.AuthPolicy); ok {
+						return true
+					}
+					if _, ok := policy.(*kuadrantv1.RateLimitPolicy); ok {
+						return true
+					}
+					return false
+				})
+				if len(policies) > 0 {
+					anyAttachedAuthorRateLimitPolicy = true
+					break outer
+				}
+			}
+		}
+
+	}
+
+	// Check that a kuadrant resource exists, and mtls enabled, and that there is at least one RateLimit or Auth Policy attached.
+	kObj := GetKuadrantFromTopology(topology)
+	if kObj == nil || !kObj.Spec.MTLS.Enable || anyAttachedAuthorRateLimitPolicy == false {
+		defer logger.V(1).Info("mtls not enabled, finishing", "status", "completed")
+		peerAuthentications := lo.FilterMap(topology.Objects().Items(), func(item machinery.Object, _ int) (machinery.Object, bool) {
+			if peerAuthentication, ok := item.(*PeerAuthentication); ok {
+				if value, exists := peerAuthentication.Labels["kuadrant.io/managed"]; exists && value == "true" {
```

Comment:
```suggestion
				if value, exists := peerAuthentication.Labels[kuadrantManagedLabelKey]; exists && value == "true" {
```
nit: `kuadrantManagedLabelKey` can be used instead here

Reply:
ack